src/mkindex_algo.hpp

```diff
@@ -630,34 +635,34 @@ auto parseAndStoreTaxTree(std::vector<bool>          & taxIdIsPresent,
 }
 
 
-template <bool is_bi,
+template <DbIndexType index_t,
           typename TStringSet>
 auto generateIndex(TStringSet                       & seqs,
                    LambdaIndexerOptions       const & options)
 {
     using TRedAlph       = seqan3::range_innermost_value_t<TStringSet>;
+    using TIndexSpec     = IndexSpec<index_t, seqan3::alphabet_size<TRedAlph>>;
+    using TIndex         = typename TIndexSpec::index_type;
 
-    constexpr auto is_collection = seqan3::text_layout::collection;
-    using TSpec = IndexSpec<seqan3::alphabet_size<TRedAlph>>;
-    using TIndex = std::conditional_t<is_bi,
-                                      seqan3::bi_fm_index<TRedAlph, is_collection, TSpec>,
-                                      seqan3::fm_index<TRedAlph, is_collection, TSpec>>;
```

Comment:
same here, I don't think this needs to be changed since we only have two index types

Reply:
reverted (adjusted a little, so it is closer to the code in shared_definitions.hpp)